src/main/kotlin/no/nav/syfo/altinnmottak/FollowUpPlanSendingService.kt

```diff
@@ -0,0 +1,45 @@
+package no.nav.syfo.altinnmottak
+
+import no.nav.syfo.application.environment.ToggleEnv
+import no.nav.syfo.client.isdialogmelding.IsdialogmeldingClient
+import no.nav.syfo.oppfolgingsplanmottak.domain.FollowUpPlanDTO
+import no.nav.syfo.oppfolgingsplanmottak.domain.FollowUpPlanResponse
+import java.util.*
+
+class FollowUpPlanSendingService(
+    private val isdialogmeldingConsumer: IsdialogmeldingClient,
+    private val altinnLpsService: AltinnLpsService,
+    private val toggles: ToggleEnv,
+) {
+    suspend fun sendFollowUpPlan(
+        followUpPlanDTO: FollowUpPlanDTO,
+        uuid: UUID,
+        employerOrgnr: String,
+    ): FollowUpPlanResponse {
+        val sykmeldtFnr = followUpPlanDTO.employeeIdentificationNumber
+
+        var sentToFastlegeStatus: Boolean? = null
+        var sentToNavStatus: Boolean? = null
+
+        if (toggles.sendLpsPlanToFastlegeToggle && followUpPlanDTO.sendPlanToGeneralPractitioner) {
+            // TODO: send actual PDF when data model and pdfgen are updated
+            sentToFastlegeStatus = isdialogmeldingConsumer.sendLpsPlanToGeneralPractitioner(sykmeldtFnr, "<MOCK PDF CONTENT>".toByteArray())
+        }
+
+        if (toggles.sendLpsPlanToNavToggle && followUpPlanDTO.sendPlanToNav) {
+            altinnLpsService.sendLpsPlanToNav(
```

Comment:
Bør kanskje flytte sendLpsPlanToNav ut av altinnLpsService da det er felles funksjon. Det ser også ut som at den kaller en funksjon database.setSentToNavTrue(uuid) som oppdaterer tabellen ALTINN_LPS. Der er en AltinnOppfolgingsplanProducer også som høres spesifikk ut til Altinn.

Så tenker enten bør vi gjøre det felles, eller lage en annen en for followupPlan

Reply:
Stemmer det. Jeg fikser.